### DISCUSSION TITLE
Configuration of DHCP server on MHI2_ER_AU57x_K3663

### DISCUSSION THREAD

**Discussion by kubierns** [Mr-MIBonk/M.I.B._More-Incredible-Bash#102] [Q&A]
Hello community,
i am new here and have a quastion.
Is it possible to set at WLAN hotspot DHCP server a gateway IP address and DNS IP address by using a custom script ? 
The default IP of DHCP server is 10.173.189.1
But clients get no gateway IP by connecting MMI hotspot.
My aim is to set gateway IP 10.173.189.254 and DNS IP 10.173.189.254, 8.8.8.8
I found something in the file
./net/mmx/mnt/system/etc/dhcp/dhcp-up
but i have no Knowledge to understand it.

My unit:
Train = MHI2_ER_AU57x_K3663
MU = MU1425
Hardware version = 054
Fazit = JTH-07621.07.17340703BD
Part number = 4G1035043J

Thanks and regards
Erni

**Comment by harman-f** [comment]
I also have not played with that so far.
MMI should share Internet via SIM card or what is you aim?

@andrewleech: Have you looked into this at some point?

Looking into `/net/mmx/mnt/system/etc/dhcp/dhcp-up` this does not look too bad:

````
if [ -z $NAMESERVER1 ]; then
    if [ -z $NAMESERVER2 ]; then
      echo "nameserver 8.8.4.4\n" >/tmp/dhcp.resolv.conf
      /bin/chmod a+rw /tmp/dhcp.resolv.conf
      /mnt/app/armle/sbin/pfctl -t dns_server -T add 8.8.4.4
      info "DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY and no DNS. Using DNS1=8.8.4.4"
      echo "*** DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY and no DNS. Using DNS1=8.8.4.4 ***" >>/dev/ser1
    else
      echo "nameserver $NAMESERVER2\n" >/tmp/dhcp.resolv.conf
      /bin/chmod a+rw /tmp/dhcp.resolv.conf
      /mnt/app/armle/sbin/pfctl -t dns_server -T add $NAMESERVER2
      info "DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS2=$NAMESERVER2"
      echo "*** DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS2=$NAMESERVER2 ***" >>/dev/ser1
    fi
  else
    if [ -z $NAMESERVER2 ]; then
      echo "nameserver $NAMESERVER1\n" >/tmp/dhcp.resolv.conf
      /bin/chmod a+rw /tmp/dhcp.resolv.conf
      /mnt/app/armle/sbin/pfctl -t dns_server -T add $NAMESERVER1
      info "DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS1=$NAMESERVER1"
      echo "*** DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS1=$NAMESERVER1 ***" >>/dev/ser1
    else
      echo "nameserver $NAMESERVER1\nnameserver $NAMESERVER2\n" >/tmp/dhcp.resolv.conf
      /bin/chmod a+rw /tmp/dhcp.resolv.conf
      /mnt/app/armle/sbin/pfctl -t dns_server -T add $NAMESERVER1 $NAMESERVER2
      info "DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS1=$NAMESERVER1 DNS2=$NAMESERVER2"
      echo "*** DHCP: $INTERFACE up with IP=$IPADDRESS MASK=$NETMASK GATEWAY=$GATEWAY DNS1=$NAMESERVER1 DNS2=$NAMESERVER2 ***" >>/dev/ser1
    fi
  fi
fi
```

**Reply by andrewleech** [reply]
I haven't looked at it myself no, I don't have any units with their own Sim card so have to share internet from phone to car

**Reply by kubierns** [reply]
Hello, 
thanks for quick answer. 
My MMI has no SIM Module and my aim is to connect MMI to Internet through OpenWrt Router in client mode. 
The network runs already well but only the MMI doesn't know the way to Internet. 
If i set gateway manually by the client internt is available.
Mayby is it possible to switch off the MMI DHCP server as workaround. 
In this case MMI has no Internet but der MMI Hotspot could provide Internet to clients.
Regards
Erni

**Reply by harman-f** [reply]
The MMI itself can directly connect to a Wifi network and use that connection. hence you can adjust DNS, IP via your router...

**Reply by kubierns** [reply]
Hello folks,
after reading a lot of log-, shell files and manuals, i found a solution.
Wlan Access Point DHCP server of MMI is using a dnsmasq server wich is configure in dnsmasq.conf file.
I made the folow changes:
login as root using D-Link DUB-E100 via putty telnet 172.16.250.248 port 123
wlan AP DHCP server is dnsmasq server wich is configure in dnsmasq.conf file.
 
make /net/mmx/mnt/system/ writeable
mount -uw /net/mmx/mnt/system/

make sd cart writeable
mount -uw /net/mmx/fs/sda0/

copy original file dnsmasq.conf to backup file dnsmasq.conf.orig
cp /net/mmx/mnt/system/etc/dnsmasq.conf /net/mmx/mnt/system/etc/dnsmasq.conf.orig

 edit dnsmasq.conf file using vi editor or copy to SD-card /net/mmx/fs/sda0/ and edit using laptop.
vi /net/mmx/mnt/system/etc/dnsmasq.conf
or
cp /net/mmx/mnt/system/etc/dnsmasq.conf  /net/mmx/fs/sda0/dnsmasq.conf
 edit using laptop

 at position after option "dhcp-range=" insert this 4 lines

dhcp-option-force=3,10.173.189.254
dhcp-option-force=6,10.173.189.254,8.8.8.8
no-resolv  
no-poll

copy from sd card to directory
cp /net/mmx/fs/sda0/dnsmasq.conf /net/mmx/mnt/system/etc/dnsmasq.conf

make /net/mmx/mnt/system/ read only
mount -ur /net/mmx/mnt/system/

Now gets very client conected to MMI a gatway IP Address where a usb modem provides internet.

regards

**Reply by rudkovss** [reply]
> The MMI itself can directly connect to a Wifi network and use that connection. hence you can adjust DNS, IP via your router...



Hey! First of all, thank you for your work with M.I.B! 
I was searching info for a while, but didn't get it.. How could MMI be connected to the iPhone as a modem?

**Reply by tbilic14** [reply]
> The MMI itself can directly connect to a Wifi network and use that connection. hence you can adjust DNS, IP via your router...

Can you help how MMI can connect to a Wifi network?

**Reply by Siskolamer** [reply]
I have same issue i and find this https://audi-intelligence.com/bluetooth/prod/ 
only android phones can share internet whit BT on Audi Cars
